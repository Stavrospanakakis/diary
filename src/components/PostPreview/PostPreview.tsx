import React from 'react'
import { Link } from 'gatsby'

interface IPostPreviewProps {
    title: string
    description: string
    date: string
    path: string
}

const PostPreview: React.FC<IPostPreviewProps> = props => {
    const { title, description, date, path } = props
    return (
        <div className="post">
            <h3 className="post__title">{ title }</h3>
            <p className="post__description">{ description }</p>
            <p className="post__date">{ date }</p>
            <Link to={ path }>Read more</Link>
        </div>
    )
}

export default PostPreview