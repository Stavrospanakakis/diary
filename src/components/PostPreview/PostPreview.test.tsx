import React from 'react'

import { render } from '@testing-library/react'

import PostPreview from './PostPreview'

describe('Navbar', () => {
	it('should render', () => {
		const { container } = render(
			<PostPreview
				image='test_image.jpg'
				title='test title'
				description='test description'
				path='test path'
				tags={['these', 'are', 'test', 'tags']}
			/>)
		expect(container).toMatchSnapshot()
	})
})

