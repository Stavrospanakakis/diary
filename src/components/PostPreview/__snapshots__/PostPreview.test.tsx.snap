// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render 1`] = `
<div>
  <div
    class="max-w-sm rounded overflow-hidden shadow-lg mt-10"
  >
    <img
      alt="test title"
      class="post-preview__image"
      src="test_image.jpg"
    />
    <h3
      class="px-6 py-4 font-bold text-purple-500 text-xl uppercase"
    >
      test title
    </h3>
    <hr
      class="py-2"
    />
    <div
      class="px-6 mx-auto"
    >
      <span
        class="inline-block bg-gray-200 rounded-full text-sm font-semibold text-gray-700 mr-2"
      >
        these
      </span>
      <span
        class="inline-block bg-gray-200 rounded-full text-sm font-semibold text-gray-700 mr-2"
      >
        are
      </span>
      <span
        class="inline-block bg-gray-200 rounded-full text-sm font-semibold text-gray-700 mr-2"
      >
        test
      </span>
      <span
        class="inline-block bg-gray-200 rounded-full text-sm font-semibold text-gray-700 mr-2"
      >
        tags
      </span>
    </div>
    <p
      class="px-6 py-4 text-gray-700 text-base text-justify"
    >
      test description
    </p>
    <a
      class="max-w-sm px-4 py-4 float-right"
      href="/test path"
    >
      <button
        class="bg-transparent hover:bg-purple-500 text-purple-900 font-semibold hover:text-white py-2 px-2 border border-purple-500 hover:border-transparent rounded"
      >
        Read More
      </button>
    </a>
  </div>
</div>
`;
